soc/arm64/nxp_imx/imx8m/soc.h

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2020 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SOC_H
+#define SOC_H
+
+#include <sys/util.h>
+#include <toolchain.h>
+
+/* TODO: Place Holder */
```

Comment:
Why are these required? Likely some other files not including what they need?

Reply:
I just follow other arm64 platforms. I think soc.h is a central place for soc related header files, currently we have not add specific device support, so leave todo for now.

Reply:
> I just follow other arm64 platforms. I think soc.h is a central place for soc related header files, currently we have not add specific device support, so leave todo for now.

That doesn't mean others do it right, really. Ref https://github.com/zephyrproject-rtos/zephyr/issues/41543

Reply:
After a check, "arch/arm64/core/smp.c" includes soc.h, so could we refactor the header file later?

Reply:
Yes, fair enough. Could be part of #41543 work.